XSD updates: Spawngroup / server handlers

baughj opened the pull request:
This PR adds `<spawngroup>` as a new tag that defines spawn groups of mobs and is intended to live in `xml/creatures/spawngroups`. Loot tables can live either in individual spawngroup XML files, or in `xml/creatures/loot`, and base mobs go in `xml/creatures/mobs`.

This PR also adds `<handlers>` to the server configuration and `<death>` as its first handler, to tell the server what to do on player death. I am adding this in anticipation of removing all the hardcoded defaults I currently have on my branch.

Lastly, any map-based definition of spawns is removed.

@anyweez @norrismiv any additions I didn't remember to add here?

anyweez commented:
Looks good overall. I didn't look too closely at loot since I haven't given that as much brainpower yet. I have some work-in-progress examples based off of these XSD's that I just built here: https://github.com/anyweez/hybrasyl.server/tree/sample-data/examples/XML/creatures. I haven't been able to validate them yet but I think they're close in case it helps to visualize what's going on here.

 A few comments, mostly cleanup. 

- To confirm: we don't want any of the `Spawn` defaults defined on the Mob, is that correct? You don't have them now but I know we discussed it.

- You mentioned wanted to convert `scriptname` => `script` everywhere. Still a few instances of `scriptname` in other XSD's so we may want to file a bug if you don't want to knock those out now. [Items](https://github.com/baughj/sdk/blob/8b7d0391e72d022e275f4537a6365f5a99d6ade1/HybrasylIntegration/HybrasylXML/XSD/Items.xsd#L235) and [castables](https://github.com/baughj/sdk/blob/8b7d0391e72d022e275f4537a6365f5a99d6ade1/HybrasylIntegration/HybrasylXML/XSD/Castables.xsd#L305) have scriptname still, for example.

- `MapList` and `SpawnList` are structured differently, which seems strange. Any reason not to make MapList.maps => MapList.map and have maxOccurs = unbounded?

- Map.map => Map.name

- Type Spawns and SpawnList are identical. Looks like Spawns can go away.

- Should Stats.hp be optional? I can see how many stats could be but HP seems inherent to all creatures.

- LootList.gold is just a uint instead of a LootGold. Shouldn't it be a LootGold?

baughj commented:
OK - `s/scriptname/script` done everywhere except Creatures.cs which will be redone later. `Maplist/Spawnlist` fixed, hp is now required.

My understanding is that yes mob is as simple as we expect. @norrismiv - is this right?

`LootList.gold` is indeed correct as we can have both:

```xml
<loot xp="3" gold="3">
````

and also
```xml
<table rolls="2" probability="0.025">
  <gold minqty="6" maxqty="1000">
</table>
```

`LootGold` represents the latter.

norrismiv commented:
Yes, mobs are simple definitions, and should only be used as references. They do not need to contain any extra information, and will allow for easier future implementations.

Loot and table look good.

@baughj where is the castables update??

anyweez commented:
👍  from me then, nice work guys!

baughj commented:
Added `usetype` as a replacement for `isclick`, and spell use enum. I changed `ChooseTarget` to simply `target` because i believe it's more consistent with `target/notarget`.

norrismiv commented:
nice.
